src/replication.c

```diff
@@ -712,6 +712,36 @@ void syncCommand(client *c) {
     /* ignore SYNC if already slave or in monitor mode */
     if (c->flags & CLIENT_SLAVE) return;
 
+    /* Check if this is a failover request to a replica with the same replid and
+     * become a master if so. */
+    if (c->argc > 3 && !strcasecmp(c->argv[0]->ptr,"psync") && 
+        !strcasecmp(c->argv[3]->ptr,"failover"))
+    {
+        serverLog(LL_WARNING, "Failover request received for replid %s.",
+            (unsigned char *)c->argv[1]->ptr);
+        if (!server.masterhost) {
+            addReplyError(c, "PSYNC FAILOVER can't be sent to a master.");
+            return;
+        }
+
+        if (!strcasecmp(c->argv[1]->ptr,server.replid)) {
+            replicationUnsetMaster();
+            sds client = catClientInfoString(sdsempty(),c);
+            serverLog(LL_NOTICE,
+                "MASTER MODE enabled (failover request from '%s')",client);
+            sdsfree(client);
```

Comment:
@oranagra I feel there is a missing return here?

Reply:
@enjoy-binbin i don't think there's a return missing, but feel free to show me where i'm wrong (i didn't debug it).

keep in mind two things:
1. PSYNC is not intended to be used from redis-cli, it's a command one redis server sends to another, not a user command.
2. PSYNC FAILOVER is a command sent by a master to one of its replicas, it does two things
    - promote the replica to be a master (few lines above this comment).
    - the old master (the one sent the PSYNC command) attempts to psync from that replica (which is why we need to keep running the bottom part of this function (the return will break it)